source/guide_tarsnap.rst

```diff
@@ -0,0 +1,232 @@
+ [isabell@stardust tarsnap-autoconf-1.0.39]$ make
+ [isabell@stardust tarsnap-autoconf-1.0.39]$ make install
+ [isabell@stardust tarsnap-autoconf-1.0.39]$ cd ..
+ [isabell@stardust ~]$ rm -rf tarsnap-autoconf-1.0.39
+ [isabell@stardust ~]$ rm tarsnap-autoconf-1.0.39.tgz
+ [isabell@stardust ~]$
+
+Configuration
+=============
+
+Create a working directory
+--------------------------
+
+.. code-block:: console
+ :emphasize-lines: 1,2
+
+ [isabell@stardust ~]$ mkdir /home/$USER/tarsnap
+ [isabell@stardust ~]$
+ 
+
+Create a user account and preload money to it
+---------------------------------------------
+
+Go to https://www.tarsnap.com/register.cgi and register an account. After that load some money to it. At least $5.
+
+
+Create some keyfiles
+--------------------
+
+.. code-block:: console
+ :emphasize-lines: 1,2
+
+ [isabell@stardust ~]$ tarsnap-keygen \
+							--keyfile /home/$USER/tarsnap/tarsnap.key \
+							--user your_registered_email_from_the_account_registration \
+							--machine some_nifty_name
+							--passphrased
+ [isabell@stardust ~]$
+
+As this key has all rights to manage your backups, you have to create a key with write-only access to do backups automatically per cronjob.
+
+.. code-block:: console
+ :emphasize-lines: 1,2
+
+ [isabell@stardust ~]$ tarsnap-keymgmt \
+							--outkeyfile /home/$USER/tarsnap/tarsnapwrite.key \
+							-w /home/$USER/tarsnap/tarsnap.key``
+ [isabell@stardust ~]$
+
+.. warning:: Please copy your keyfiles to a safe place! Without the keys you cannot access your backups anymore.
+
+
+Set up the config file
+----------------------
+
+.. code-block:: console
+ :emphasize-lines: 1,2
+
+ [isabell@stardust ~]$ cp /home/$USER/etc/tarsnap.conf.sample /home/$USER/etc/tarsnap.conf
+ [isabell@stardust ~]$
+
+Now edit ``/home/$USER/etc/tarsnap.conf`` with the editor of your choice. The config file is already commented by the author.
+Make sure to use your write-only keyfile under keyfile in the config file.
+
+
+Create a backup script
+----------------------
+
+Create a script ``/home/$USER/tarsnap-backup.sh`` using the editor of your choice with the following content:
+
+``!/bin/sh
+/home/$USER/bin/tarsnap -c \
+	-f "BACKUP-$(date +%d-%m-%Y_%H-%M-%S)" \
+	/var/www/virtual/$USER``
+
+Now make it executable.
+
+.. code-block:: console
+ :emphasize-lines: 1,2
+
+ [isabell@stardust ~]$ chmod u+x /home/$USER/tarsnap-backup.sh
+ [isabell@stardust ~]$
+
+
+Setup automatic backups per cronjob
+-----------------------------------
+
+.. code-block:: console
+ :emphasize-lines: 1,2
+
+ [isabell@stardust ~]$ crontab -e
+ [isabell@stardust ~]$
+
+Enter ``0 02 * * * /home/$USER/tarsnap-backup.sh`` to let the backup run every night at 2 am.
+
+.. hint:: For help setting up cronjobs go to https://crontab.guru/
+
+
+Test your backup
+----------------
+
+Start a backup using
+
+.. code-block:: console
+ :emphasize-lines: 1,2
+
+ [isabell@stardust ~]$ /home/$USER/tarsnap-backup.sh
```

Comment:
This fails for me:
```
[tsnap@lynx ~]$ ./tarsnap-backup.sh 
tarsnap: Cannot create directory: /usr/local/tarsnap-cache: Permission denied
tarsnap: Failed to ensure that cachedir exists
```
Maybe the `cachedir` setting in the config file needs a new path like `/home/isabell/tmp`?

Reply:
Hm, sure you entered the cache directory correctly in the config? In your example the path is "tsnap" instead of "tarsnap". But I added a line to create the cache directory just to be sure, even though I'm almost sure that tarsnap created the cache directory itself if a tarsnap directory was created manually before.

The other things should now been corrected or added. Thanks for the help. I've never done anything like this before and found it quite difficult as I couldn't really see the result of all the writing. I was able to see it for the first time in the deploy preview you linked. Is there actually an editor (online or on the computer) where you can see the layout? Like WYSIWYG? :o)